src/Symfony/Component/Lock/Tests/Store/MongoDbStoreTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+use MongoDB\Client;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Store\MongoDbStore;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * @author Joe Bennett <joe@assimtech.com>
+ * @requires extension mongodb
+ */
+class MongoDbStoreTest extends AbstractStoreTest
+{
+    use ExpiringStoreTestTrait;
+
+    public static function setupBeforeClass()
+    {
+        $client = self::getMongoConnection();
+        $client->listDatabases();
+    }
+
+    /**
+     * @return Client
+     */
+    private static function getMongoConnection(): Client
+    {
+        return new Client('mongodb://'.getenv('MONGODB_HOST'));
+    }
+
+    /**
+     * @return int
+     */
+    protected function getClockDelay(): int
+    {
+        return 250000;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStore(): StoreInterface
+    {
+        return new MongoDbStore(self::getMongoConnection(), array(
+            'database' => 'test',
+        ));
+    }
+
+    public function testCreateIndex()
+    {
+        $store = $this->getStore();
+
+        $this->assertEquals($store->createTtlIndex(), 'expires_at_1');
+    }
+
+    public function testNonBlocking()
+    {
+        $store = $this->getStore();
+
+        $key = new Key(uniqid(__METHOD__, true));
+
+        try {
+            $store->waitAndSave($key);
+            $this->fail('The store shouldn\'t support waitAndSave');
+        } catch (NotSupportedException $e) {
```

Comment:
use https://phpunit.de/manual/6.5/en/appendixes.annotations.html#appendixes.annotations.expectedException to assert that an exception is thorwn

Reply:
updated